<launch>

	<node name="streamer" pkg="thermalvis" type="streamer">
	
		<param name="debugMode" value="true" />
	
		<param name="source" value="directory" />
		<param name="directory" value="~\Documents\Data\optris\low-snr" /> -->
		<!-- <param name="directory" value="C:\Users\Stephen\Documents\Data\optris\Denoised sequence\Non-local method\low_snr" /> -->
		<!-- <param name="directory" value="C:\Users\Stephen\Documents\Data\optris\Denoised sequence\Transform-Domain Collaborative Filtering\low_snr" /> -->
		
		<param name="inputDatatype" value="1" /> <!-- 0 for 8-bit, 1 for raw (14-bit), 2 for multi-modality (2 channel), 3 for depth (16-bit) -->
		
		<param name="markDuplicates" value="true" />
		
		<param name="filterMode" value="0" /> <!-- 0 for no filter, 1 for gaussian, 2 for bilateral, 3 for adaptive bilateral -->
		<param name="filterParam" value="2.0" />
		
		<!-- <param name="degreesPerGraylevel" value="0.025" /> --> <!-- value of 0.025 works OK for miricle sequences.. -->
		<param name="desiredDegreesPerGraylevel" value="0.05" /> <!-- can adjust this to enhance contrast -->
		
		<param name="outputType" value="0" /> <!-- 0 for 8-bit grayscale, 1 for 8-bit color, 2 for 16-bit grayscale -->
		<param name="writeImages" value="true" />
		<param name="outputFolder" value="~\Documents\Data\optris\output" />
		
		<param name="loopMode" value="false" />
		
		<param name="outputColor" value="false" />
		<param name="output8bit" value="true" />

	</node>
	
	<node name="image_view" pkg="image_view" type="image_view" output="screen">
		<remap from="image" to="/thermalvis/streamer/image_mono"/>
		<param name="~/autosize" value="true" />
	</node>
	
	<node name="reconfigure_flow" pkg="dynamic_reconfigure" type="reconfigure_gui" output="screen" args="/streamer"/>
	
</launch>
